lms/djangoapps/courseware/utils.py

```diff
@@ -121,38 +121,46 @@ def is_eligible_for_financial_aid(course_id):
-        elif response.status_code == status.HTTP_400_BAD_REQUEST:
-            return False, response.json().get('message')
+        return client.request(method, f"{financial_assistance_configuration.api_url}{url}", params=params, data=data)
     else:
         return False, 'Financial Assistance configuration is not enabled'
 
 
+def is_eligible_for_financial_aid(course_id):
+    """
+    Sends a get request to edx-financial-assistance to retrieve financial assistance eligibility criteria for a course.
+
+    Returns either True if course is eligible for financial aid and vice versa.
+    Also returns the reason why the course isn't eligible.
+    In case of a bad request, returns an error message.
+    """
+    response = _call_edx_financial_assistance('GET', f"{settings.IS_ELIGIBLE_FOR_FINANCIAL_ASSISTANCE_URL}{course_id}/")
+    if response.status_code == status.HTTP_200_OK:
+        return response.json().get('is_eligible'), response.json().get('reason')
+    elif response.status_code == status.HTTP_400_BAD_REQUEST:
+        return False, response.json().get('message')
+    else:
+        return False, UNEXPECTED_ERROR_IS_ELIGIBLE
+
+
 def get_financial_assistance_application_status(user_id, course_id):
     """
     Given the course_id, sends a get request to edx-financial-assistance to retrieve
-    financial assistance application status for the logged-in user.
+    financial assistance application(s) status for the logged-in user.
     """
-    application_status_url = f"/core/api/financial_assistance_application/status/?" \
-                             f"{parse.urlencode({'course_id': course_id})}&{parse.urlencode({'lms_user_id': user_id})}"
-    financial_assistance_configuration = FinancialAssistanceConfiguration.current()
-    if financial_assistance_configuration.enabled:
-        oauth_application = Application.objects.get(user=financial_assistance_configuration.get_service_user())
-        client = OAuthAPIClient(
-            settings.LMS_ROOT_URL,
-            oauth_application.client_id,
-            oauth_application.client_secret
-        )
-        response = client.request('GET', f"{financial_assistance_configuration.api_url}{application_status_url}")
-        if response.status_code == status.HTTP_200_OK:
-            return True, response.json()
-        elif response.status_code in (status.HTTP_400_BAD_REQUEST, status.HTTP_404_NOT_FOUND):
-            return False, response.json().get('message')
-
+    request_params = {
+        'course_id': course_id,
+        'lms_user_id': user_id
+    }
+    response = _call_edx_financial_assistance(
+        'GET', f"{settings.FINANCIAL_ASSISTANCE_APPLICATION_STATUS_URL}", params=request_params
+    )
+    if response.status_code == status.HTTP_200_OK:
+        return True, response.json()
+    elif response.status_code in (status.HTTP_400_BAD_REQUEST, status.HTTP_404_NOT_FOUND):
+        return False, response.json().get('message')
```

Comment:
have a last else where you should log the response, this will help in debugging in case of an issue